Infrastructure/Repositories/TaskRepository.cs

```diff
@@ -91,18 +57,16 @@ public async Task<TaskModel> CreateAsync(TaskModel task)
         {
             task.Id = Guid.NewGuid();
         }
-
-        await Task.Run(() =>
-        {
-            _tasks.Add(task);
-            project.Tasks.Add(task);
-        });
+        dbContext.Tasks.Add(task);
+        project.Tasks.Add(task);
+        await dbContext.SaveChangesAsync();
         return task;
     }
 
     public async Task<TaskModel> UpdateAsync(TaskModel task)
     {
         var existingTask = await GetByIdAsync(task.Id);
```

Comment:
Why don't you use the dbContext.Tasks.Update(entity) method?

Reply:
I dont quite understand how we can use it in repository

Reply:
Please read it:
https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.dbset-1.update?view=efcore-8.0

Reply:
fixed

Reply:
I don't agree, you should update task from method argument, not existing task